js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetsGroupedView.tsx

```diff
@@ -0,0 +1,540 @@
+                    link: linkToAssetTableWithAssetOwnerFilter(owner),
+                  };
+                  acc.owners[owner.team]!.assets.push(asset);
+                  break;
+                case 'UserAssetOwner':
+                  acc.owners[owner.email] = acc.owners[owner.email] || {
+                    assets: [],
+                    label: owner.email,
+                    link: linkToAssetTableWithAssetOwnerFilter(owner),
+                  };
+                  acc.owners[owner.email]!.assets.push(asset);
+                  break;
+                default:
+                  assertUnreachable(owner);
+              }
+            });
+          }
+          if (groupName) {
+            acc.groupName[groupName] = acc.groupName[groupName] || {
+              assets: [],
+              label: groupName,
+              link: linkToAssetTableWithGroupFilter({
+                groupName,
+                repositoryLocationName: repository?.location.name,
+                repositoryName: repository?.name,
+              }),
+            };
+            acc.groupName[groupName]!.assets.push(asset);
+          }
+          if (repository) {
+            const name = buildRepoPathForHuman(repository.name, repository.location.name);
+            acc.repository[name] = acc.repository[name] || {
+              assets: [],
+              label: name,
+              link: linkToCodeLocation(buildRepoAddress(repository.name, repository.location.name)),
+            };
+            acc.repository[name]!.assets.push(asset);
+          }
+          if (tags) {
+            tags.forEach((tag) => {
+              const stringValue = `${tag.key}${tag.value ? ': ' + tag.value : ''}`;
+              acc.tags[stringValue] = acc.tags[stringValue] || {
+                assets: [],
+                label: stringValue,
+                link: linkToAssetTableWithTagFilter(tag),
+              };
+              acc.tags[stringValue]!.assets.push(asset);
+            });
+          }
+          if (kinds) {
+            kinds.forEach((kind) => {
+              acc.kinds[kind] = acc.kinds[kind] || {
+                assets: [],
+                label: kind,
+                link: linkToAssetTableWithKindFilter(kind),
+              };
+              acc.kinds[kind]!.assets.push(asset);
+            });
+          }
+          return acc;
+        },
+        {
+          owners: {} as Record<string, {label: string; assets: AssetTableFragment[]; link: string}>,
+          groupName: {} as Record<
+            string,
+            {label: string; assets: AssetTableFragment[]; link: string}
+          >,
+          repository: {} as Record<
+            string,
+            {label: string; assets: AssetTableFragment[]; link: string}
+          >,
+          tags: {} as Record<string, {label: string; assets: AssetTableFragment[]; link: string}>,
+          kinds: {} as Record<string, {label: string; assets: AssetTableFragment[]; link: string}>,
```

Comment:
Define this type in one place for reuse?

Reply:
Ah yeah let me do that